![top-banner](https://raw.githubusercontent.com/PS43Foundation/data-dunkers/main/docs/top-banner.jpg)

<a target="_parent" href="https://colab.research.google.com/github/PS43Foundation/data-dunkers/blob/main/notebooks/hackathon/pascal-siakam-stats-challenge.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/>
</a>

<a href="https://hub.callysto.ca/jupyter/hub/user-redirect/git-pull?repo=https%3A%2F%2Fgithub.com%2FPS43Foundation%2Fdata-dunkers&branch=main&subPath=notebooks/hackathon/pascal-siakam-stats-challenge.ipynb&depth=1" target="_parent"><img src="https://raw.githubusercontent.com/callysto/curriculum-notebooks/master/open-in-callysto-button.svg?sanitize=true" width="123" height="24" alt="Open in Callysto"/></a>

<a target="_parent" href="https://2i2c.callysto.ca/hub/user-redirect/git-pull?repo=https%3A%2F%2Fgithub.com%2FPS43Foundation%2Fdata-dunkers&branch=main&subPath=notebooks/hackathon/pascal-siakam-stats-challenge.ipynb&depth=1"><img src="https://2i2c.org/media/logo.svg" width="30" height="30" alt="Open in 2i2c Hub"/></a>

# Pascal Siakam Statistics

Submitted by: Dumitru Gliga and Noah Walsh

## Introduction

Pascal Siakam took an unconventional road to stardom. He didn't grow up playing basketball, but instead began only when he was in high school.

In this notebook we will use statistics from [Basektball Reference](https://www.basketball-reference.com).

## Challenges

1. Plot some of Pascal Siakam's statistics over time, such as points per game or rebounds per game. Are there any patterns?
1. Pascal has taken on more of a leadership role with the Toronto Raptors since they won a championship in 2019, meaning that he has more responsibility, plays more minutes, and takes more shots. Plot a graph that contains some or all of the previous stats, but also contains his Minutes Per Game. What does this tell us about how Pascal is playing now that he is the "star" of the team?
1. We can also take a look at how "efficient" he is as he evolves by looking at his shooting percentages for different types of shots (we’ll start with 2-pointers and 3-pointers). Plot his shooting percentage by year for his career. What trends do you notice?
1. How does Pascal Siakam's efficiency change as he becomes a superstar? Plot a graph containing his Minutes Per Game, 2 Pt % and 3 Pt % over time. What do you notice? Does a higher usage lead to a greater or lesser efficiency?
1. If you were the General Manager and looking at these trends, what would you prioritize to maximize the talents of Pascal Siakam?
1. Using the 2016 Draft Class, plot a graph of the Total Career Points of the first round draft picks. Is Plascal Siakam near the top?
1. Find the top 5 total point scorers of that draft class. For each, find their player page and then plot their Points Per Game for their careers thus far.

In [ ]:
import pandas as pd
import plotly.express as px
per_game = pd.read_html('https://www.basketball-reference.com/players/s/siakapa01.html')[2]
draft_class = pd.read_html('https://www.basketball-reference.com/draft/NBA_2016.html')[0]
per_gameHB = pd.read_html ('https://www.basketball-reference.com/players/h/hieldbu01.html')[2]
per_gameJB = pd.read_html ('https://www.basketball-reference.com/players/b/brownja02.html')[2]
per_gameBI = pd.read_html ('https://www.basketball-reference.com/players/i/ingrabr01.html#per_game')[1]
per_gameDS = pd.read_html ('https://www.basketball-reference.com/players/s/sabondo01.html')[2]
draft_class = draft_class.fillna(0)
draft_class = draft_class.drop([30, 31]).reset_index(drop=True)
draft_class.columns = draft_class.columns.droplevel()
draft_class.columns = ['Rk','Pk','Tm','Player','College','Yrs','G','MP','PTS','TRB','AST','FG%','3P%','FT%','MPPG','PTSPG','TRBPG','ASTPG','WS','WS/48','BPM','VORP']
for column in draft_class.columns:
    draft_class[column] = pd.to_numeric(draft_class[column], errors='ignore')
print('Libraries and data imported')
print(per_game.head())

Libraries and data imported
    Season   Age   Tm   Lg Pos   G  GS    MP   FG   FGA  ...    FT%  ORB  DRB  \
0  2016-17  22.0  TOR  NBA  PF  55  38  15.6  1.9   3.7  ...  0.688  1.2  2.2   
1  2017-18  23.0  TOR  NBA  PF  81   5  20.7  3.1   6.1  ...  0.621  1.0  3.5   
2  2018-19  24.0  TOR  NBA  PF  80  79  31.9  6.5  11.8  ...  0.785  1.6  5.3   
3  2019-20  25.0  TOR  NBA  PF  60  60  35.2  8.3  18.4  ...  0.792  1.1  6.3   
4  2020-21  26.0  TOR  NBA  PF  56  56  35.8  7.8  17.2  ...  0.827  1.7  5.5   

   TRB  AST  STL  BLK  TOV   PF   PTS  
0  3.4  0.3  0.5  0.8  0.6  2.0   4.2  
1  4.5  2.0  0.8  0.5  0.8  2.0   7.3  
2  6.9  3.1  0.9  0.7  1.9  3.0  16.9  
3  7.3  3.5  1.0  0.9  2.5  2.8  22.9  
4  7.2  4.5  1.1  0.7  2.3  3.1  21.4  

[5 rows x 30 columns]


In [ ]:
per_gameHB.drop_duplicates("Season", keep="first").reset_index(drop=True)

Season   Age   Tm   Lg  Pos      G     GS    MP   FG   FGA  ...    FT%  \
0     2016-17  24.0  TOT  NBA   SG   82.0   55.0  23.0  4.0   9.4  ...  0.842   
1     2017-18  25.0  SAC  NBA   SG   80.0   12.0  25.3  5.2  11.7  ...  0.877   
2     2018-19  26.0  SAC  NBA   SG   82.0   82.0  31.9  7.6  16.6  ...  0.886   
3     2019-20  27.0  SAC  NBA   SG   72.0   44.0  30.8  6.9  16.1  ...  0.846   
4     2020-21  28.0  SAC  NBA   SG   71.0   71.0  34.3  5.7  14.0  ...  0.846   
5     2021-22  29.0  TOT  NBA   SG   81.0   32.0  30.9  5.5  13.6  ...  0.874   
6     2022-23  30.0  IND  NBA   SF   80.0   73.0  31.0  5.9  13.0  ...  0.822   
7     2023-24  31.0  IND  NBA   SF    5.0    0.0  23.4  4.2  10.0  ...    NaN   
8      Career   NaN  NaN  NBA  NaN  553.0  369.0  29.4  5.8  13.4  ...  0.858   
9         NaN   NaN  NaN  NaN  NaN    NaN    NaN   NaN  NaN   NaN  ...    NaN   
10  6 seasons   NaN  SAC  NBA  NaN  385.0  233.0  30.1  6.1  14.1  ...  0.863   
11  3 seasons   NaN  IND  NBA  NaN  111.0   99.0  31.7  6.1  13.5  ...  0.835   
12   1 season   NaN  NOP  NBA  NaN   57.0   37.0  20.4  3.2   8.3  ...  0.879   

    ORB  DRB  TRB  AST  STL  BLK  TOV   PF   PTS  
0   0.4  2.9  3.3  1.5  0.5  0.1  1.2  1.4  10.6  
1   0.7  3.2  3.8  1.9  1.1  0.3  1.6  1.9  13.5  
2   1.3  3.7  5.0  2.5  0.7  0.4  1.8  2.5  20.7  
3   0.8  3.8  4.6  3.0  0.9  0.2  2.3  2.3  19.2  
4   0.4  4.3  4.7  3.6  0.9  0.4  1.8  2.5  16.6  
5   0.8  3.5  4.4  2.8  0.9  0.3  1.9  2.3  15.6  
6   0.8  4.2  5.0  2.8  1.2  0.3  1.7  2.0  16.8  
7   0.8  2.6  3.4  4.0  0.4  0.6  1.2  1.2  11.2  
8   0.8  3.6  4.4  2.6  0.9  0.3  1.7  2.1  16.0  
9   NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN   NaN  
10  0.8  3.6  4.4  2.5  0.9  0.3  1.8  2.2  16.9  
11  0.8  4.1  4.9  3.3  1.1  0.4  1.9  2.1  16.9  
12  0.3  2.6  2.9  1.4  0.3  0.1  0.8  1.4   8.6  

[13 rows x 30 columns]

### Pascal Siakam Per Game Columns

|Column|Meaning|
|-|-|
|Season|If listed as single number, the year the season ended|
|Age|Player's age on February 1 of the season|
|Tm|Team|
|Lg|League|
|Pos|Position|
|G|Games|
|GS|Games Started|
|MP|Minutes Played Per Game|
|FG|Field Goals Per Game|
|FGA|Field Goal Attempts Per Game|
|FG%|Field Goal Percentage|
|3P|3-Point Field Goals Per Game|
|3PA|3-Point Field Goal Attempts Per Game|
|3P%|3-Point Field Goal Percentage|
|2P|2-Point Field Goals Per Game|
|2PA|2-Point Field Goal Attempts Per Game|
|2P%|2-Point Field Goal Percentage|
|eFG%|Effective Field Goal Percentage (since a 3-point field goal is worth more)|
|FT|Free Throws Per Game|
|FTA|Free Throw Attempts Per Game|
|FT%|Free Throw Percentage|
|ORB|Offensive Rebounds Per Game|
|DRB|Defensive Rebounds Per Game|
|TRB|Total Rebounds Per Game|
|AST|Assists Per Game|
|STL|Steals Per Game|
|BLK|Blocks Per Game|
|TOV|Turnovers Per Game|
|PF|Personal Fouls Per Game|
|PTS|Points Per Game|

### NBA 2016 Draft Class Columns

|Column|Meaning|
|-|-|
|Rk|Rank|
|Pk|Overall Pick|
|Tm|Team|
|Yrs|Seasons that player has appeared in the NBA|
|G|Games|
|MP|Minutes Played|
|PTS|Points|
|TRB|Total Rebounds|
|AST|Assists|
|FG%|Field Goal Percentage|
|3P%|3-Point Field Goal Percentage|
|FT%|Free Throw Percentage|
|MPPG|Minutes Played Per Game|
|PTSPG|Points Per Game|
|TRBPG|Total Rebounds Per Game|
|ASTPG|Assists Per Game|
|WS|Win Shares, an estimate of the number of wins contributed by a player|
|WS/48|Win Shares Per 48 Minutes (league average is about .100)|
|BPM|Box Plus/Minus, estimate of points per 100 possessions a player contributed above a league-average player, translated to an average team|
|VORP|Value over Replacement Player, estimate of points per 100 TEAM possessions a player contributed above a replacement-level (-2.0) player, translated to an average team and prorated to an 82-game season|

## Analysis



In [ ]:
px.bar(per_game, x='Age', y=['MP', 'FGA', 'FG', 'PTS'], barmode='group', title='Pascal Siakam Confidence').update_layout(yaxis_title='Percentage')

Here we can see that as he got older, more confident and just ,in general, became the star of the team (specifically in 2019 when he took more of a leadership role) he got more confident. When he got more confident, he went for more attempts which lead him to get better and thus have more field goals which directly lead to more points per game.

In [ ]:
px.scatter(per_game, x='MP', y=['2P%', '3P%'], title="Pascal Siakam's 2 point % and 3 point % relating to minutes played", color='Age')

Here we can see that as Pascal got older and became the star of the team(thus having to be a leader and play more) his 3 point % improved significantly while his 2 point % remained fairly stable. This can be attributed to multiple factors such as 3 points being more and more essential in the game of basketball (this also explains the drop in 2 point %) ,the fact that as he got to play more he got more consistent with his 3 point % and that as he played more he developped other skills which all helped him developpe his game IQ meaning that he took smarter shots.


If I were the manager I would prioritize putting Pascal in offensive positions where he constantly gets to go for more shots because as we saw Pascal is the type of Player who ,as he plays more, he evolves into a better player and adapts to the changes of the game of basketball as a whole.

****

In [ ]:
px.scatter(draft_class, x='Pk', y='PTS', hover_name='Player', title='Picks relating to total points scored', trendline='ols')

Here we would expect a fairly good correlation between overall pick and points as one would think that the top prospects would performe the best and yet, we see that this is rarely the case there are alot of outliers, such as Pascal, who is the player with the 3rd most overall points despite being the 27th pick during the 2016 draft. We also see that very little players actually reach the trend line and that instead we have alot of outliers who do much better than expected.


In [ ]:
px.line(per_gameHB, x='Age', y='PTS', title='Buddy Hill Points per Game')

In [ ]:
px.line(per_gameJB, x='Age', y='PTS', title='Jaylen Brown Points per Game')

In [ ]:
px.line(per_gameBI, x='Age', y='PTS', title='Brandon Ingram Points per Game')

In [ ]:
px.line(per_gameDS, x='Age', y='PTS', title='Domantas Sabonis Points per Game')

## Reflection Questions

* What is something you learned through this process?
* What would you like to show others?
* How can you apply your learning to future activities?

[![Callysto.ca License](https://github.com/PS43Foundation/data-dunkers/blob/main/bottom-banner.jpg?raw=1)](https://github.com/callysto/curriculum-notebooks/blob/master/LICENSE.md)